test/unit/typeahead_spec.js

```diff
@@ -368,6 +367,28 @@ describe('Typeahead', function() {
     });
   });
 
+  describe('when clearOnSelected flag is set to true', function() {
+    it('clears input when selected', function() {
+      var spy = jasmine.createSpy();
+      var view = new Typeahead({
+        input: this.$input,
+        clearOnSelected: true,
+        hint: true,
+        datasets: {}
+      });
+      view.dropdown.getDatumForCursor.and.returnValue(testDatum);
+
+
+      var $e = jasmine.createSpyObj('event', ['preventDefault']);
+      view.$input.on('autocomplete:selected', spy);
+      view.input.trigger('enterKeyed', $e);
+
+      expect(spy).toHaveBeenCalled();
+      expect(view.input.setQuery).toHaveBeenCalledWith('');
```

Comment:
Can you check the value of this.$input rather than checking that some methods have been called (in order to not test how the code is done)?
Also would it be possible to have some value to check that there is an actual change of state (this would be a requirement if we can test without spy)?

Reply:
Makes sense, but didn't know how to set a query here via code 🤔